compiler/rustc_attr/src/builtin.rs

```diff
@@ -53,13 +53,22 @@ pub enum InstructionSetAttr {
     ArmT32,
 }
 
-#[derive(Clone, Encodable, Decodable, Debug, HashStable_Generic)]
+#[derive(Clone, Encodable, Decodable, Debug, HashStable_Generic, PartialEq, Eq)]
 pub enum OptimizeAttr {
+    /// `#[optimize(none)]`
     None,
+    /// `#[optimize(speed)]`
     Speed,
+    /// `#[optimize(size)]`
     Size,
 }
 
+impl OptimizeAttr {
+    pub fn is_none(&self) -> bool {
+        matches!(*self, OptimizeAttr::None)
+    }
+}
```

Comment:
I don't like this function. It is named the same way as an `Option::is_none` but has different semantics. we should make it explicit that someone is checking for the variant `None`. On a different note, maybe we could rename this function to `do_not_optimize` instead.